wolfSSL/src/chapter02.md

```diff
@@ -680,15 +1234,325 @@ Enable ECC SECP384R1 Single-Precision (SP) support. Only applies to `WOLFSSL_SP_
 
 Enable SAKKE pairing based cryptography Single-Precision (SP) support.
 
+#### ATOMIC_USER
+
+Enable Atomic Record Layer callbacks.
+
+#### BIG_ENDIAN_ORDER
+
+Endianness - defaults to little endian, i.e., makes big-endian.
+
+#### WOLFSSL_32BIT_MILLI_TIME
+
+Function TimeNowInMilliseconds() returns an unsigned 32-bit value. The default behavior is to return a signed 64-bit value.
+
+#### WOLFSSL_MAX_DHKEY_BITS
+
+DH maximum bit size must be a multiple of 8. DH maximum bit size must not exceed 16384 or greater than WOLFSSL_MIN_DHKEY_BITS.
+
+#### WOLFSSL_MIN_DHKEY_BITS
+
+The DH minimum bit size must be a multiple of 8. For 112 bits of security, DH needs at least 2048-bit keys, and the minimum bit size must not be greater than 16000.
+
+#### WOLFSSL_MAX_MTU
+
+Max expected MTU. 1500 - 100 bytes to account for UDP and IP headers.
+
+#### IGNORE_NETSCAPE_CERT_TYPE
+
+Define to input netscape cert type but holds a place.
+
+#### SESSION_CERTS
+
+TLS session cache for certs.
+
+#### WOLFSSL_DUAL_ALG_CERTS
+
+Dual algorithm certificate required feature.
+
+#### CRL_MAX_REVOKED_CERTS
+
+Specifies the number of buffers to hold RevokedCerts. The default value is set to 4.
+
+#### CRL_STATIC_REVOKED_LIST
+
+Enables a fixed static list of RevokedCerts to allow for a binary search.
+
+#### SESSION_INDEX
+
+Identifies the session's location in the cache. Specifies index session/row shifts.
+
+#### SESSION_TICKET_HINT_DEFAULT
+
+The ticket hint default is used to set the default hint value. The Ticket Key lifetime must be longer than the ticket life hint.
+
+#### WOLFSSL_DTLS13
+
+Enable wolfSSL DTLS 1.3.
+
+#### WOLFSSL_TLS13
+
+Enable TLS 1.3 protocol implementation.
+
+#### WOLFSSL_TLS13_IGNORE_AEAD_LIMITS
+
+Limits specified by https://www.rfc-editor.org/rfc/rfc9147.html#name-aead-limits. We specify the limit by which we need to do a key update as the halfway point to the hard decryption fail limit.
+
+#### WOLFSSL_TLS13_MIDDLEBOX_COMPAT
+
+Enable middlebox compatibility in the TLS 1.3 handshake. This includes sending ChangeCipherSpec before encrypted messages and including a session ID.
+
+#### WOLFSSL_TLS13_SHA512
+
+Allow generation of SHA-512 digests in a handshake - no ciphersuite. Requires SHA-512 at this time.
```

Comment:
Per tls13.c -> Allow generation of SHA-512 digests in handshake - no ciphersuite requires SHA-512 at this time. This enables calculation of a SHA2-512 hash for the handshake messages even though its not used by TLS v1.3 yet.